File: server/src/main/java/com/hospitaltender/server/service/UserService.java
package com.hospitaltender.server.service;

import com.hospitaltender.server.dto.request.CreateUserRequest;
import com.hospitaltender.server.dto.request.UpdateUserRequest;
import com.hospitaltender.server.dto.response.UserResponse;

import java.util.List;

public interface UserService {

    UserResponse create(CreateUserRequest request);

    UserResponse getById(Long id);

    List<UserResponse> getAll();

    UserResponse update(Long id, UpdateUserRequest request);

    void delete(Long id);
}

File: ui/hospital-system-ui/src/services/user.service.ts
import api from './api'
import { UserResponse, CreateUserRequest, UpdateUserRequest } from '@/types'

export const userService = {
  getAll: () =>
    api.get<UserResponse[]>('/users').then((res) => res.data),

  getById: (id: number) =>
    api.get<UserResponse>(`/users/${id}`).then((res) => res.data),

  create: (data: CreateUserRequest) =>
    api.post<UserResponse>('/users', data).then((res) => res.data),

  update: (id: number, data: UpdateUserRequest) =>
    api.put<UserResponse>(`/users/${id}`, data).then((res) => res.data),

  delete: (id: number) =>
    api.delete(`/users/${id}`),
}
